main.go

```diff
@@ -106,11 +120,191 @@ func newVault(addr string) (vault *vapi.Client) {
 	return vault
 }
 
+// getKubeClient returns a kubernetes clientset.
+//
+// This function will attempt to get the in-cluster config. The ServiceAccount requires the
+// following permissions:
+// - `get` on `services` in all namespaces
+// - `get` on `pods` in all namespaces
+func getKubeClient() (*kubernetes.Clientset, error) {
+	kubeconfig, err := rest.InClusterConfig()
+	if err != nil {
+		logger.WithError(err).Fatal("error getting in-cluster config")
+	}
+
+	client := new(kubernetes.Clientset)
+	if client, err = kubernetes.NewForConfig(kubeconfig); err != nil {
+		logger.WithError(err).Fatal("error creating kubernetes client")
+	}
+
+	return client, nil
+}
+
+func getVaultPodsForService() ([]string, error) {
+	client, err := getKubeClient()
+	if err != nil {
+		return nil, fmt.Errorf("error getting kubernetes client: %w", err)
+	}
+
+	services, err := client.CoreV1().Services(getNamespaceFromConfig()).List(context.Background(), metav1.ListOptions{
+		LabelSelector: labels.Set{
+			appNameLabel: defaultVaultName,
+		}.String(),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting services: %w", err)
+	}
+
+	if len(services.Items) == 0 {
+		return nil, errors.New("no services found")
+	}
+
+	podAddrs := make([]string, 0)
+	for _, service := range services.Items {
+		pods, err := client.CoreV1().Pods(getNamespaceFromConfig()).List(context.Background(), metav1.ListOptions{
+			LabelSelector: labels.Set{
+				appNameLabel:  defaultVaultName,
+				instanceLabel: service.Name,
+			}.String(),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error getting pods for service %q: %w", service.Name, err)
+		}
+
+		for _, pod := range pods.Items {
+			podAddrs = append(podAddrs, getVaultAddr(service.Spec.Ports, pod.Status.PodIP))
+		}
+	}
+
+	return podAddrs, nil
+}
+
+func getVaultAddr(service []core.ServicePort, ip string) string {
+	const targetScheme = "http"
+
+	for _, port := range service {
+		if port.Name == targetScheme {
+			return fmt.Sprintf("%s://%s:%d", port.Protocol, ip, port.Port)
```

Comment:
this is causing the address to return "TCP://XXX", which is invalid

Reply:
You need to set the scheme to http